### ISSUE TITLE
Making an Eclipse Project doesn't add C++ include paths

### ISSUE THREAD

**Issue by b-adkins** [ros/catkin#551]
TL;DR: `/usr/include/c++/4.x` missing from Eclipse project when built with Catkin, please add it.

When using catkin to build an Eclipse project (See http://wiki.ros.org/IDEs#Eclipse), a number of include paths are given to Eclipse. They are found under:
`Project > Properties > C/C++ Include Paths and Symbols`

System (non-project) includes already added are:
- `/usr/include`
- `/usr/src/gtest`
- `/usr/src/gtest/include`
- `/opt/ros/groovy/include`

This is a problem because without it, Eclipse's indexer can't find any C++ library code, which are used all over ROS. Every time I have to empty the `devel` folder, I have to rebuild my Eclipse project and I add the path manually.

Even assuming an arbitrary ROS project doesn't use C++ libraries, what's the harm in telling Eclipse where to find it? It's not like Eclipse is used because it's efficient software. ;-)

**Comment by wjwwood** [comment]
This is been discussed before, I think the conclusion was that it didn't make sense to add this in catkin. Conceptually it makes no sense for catkin to add any include directories, if the generated eclipse project is not correct, then it is CMake's fault.

Previous discussions:

http://answers.ros.org/question/72226/eclipse-not-resolving-anything/
http://answers.ros.org/question/61407/standard-system-include-paths-missing-in-eclipse-from-catkin_make/
http://answers.ros.org/question/68074/eclipse-and-catkin-problems-symbol-could-not-be-resolved/

There were some potential work arounds in those discussions if I remember correctly.

Ideally you should be able to add those kinds of global include paths to eclipse above the the project configuration level, but if I remember correctly eclipse doesn't allow that. The reason I believe eclipse should be the place to add these include directories and not catkin is that they are not required to compile and tend to be platform or development environment specific. I neither think it is catkin's responsibility nor do I think it is a good idea to couple this kind of thing with catkin.

**Comment by dirk-thomas** [comment]
I have to agree with @wjwwood. This can / should not be addressed within catkin. Please consider providing the feedback to CMake / Eclipse.
